### PR TITLE
Implementation of the unit test

### PR DESCRIPTION
**Pull request by ezequielsetzesglobant**
The unit test of both the ViewModels, UseCase and Mappers was implemented.

Unit test of ViewModels and UseCase
image

Coverage of ViewModels and UseCase
image

Unit test of Mappers
image
image

Coverage of Mappers
image

### CODE REVIEW

**Review by SartiMau — APPROVED** [review]

**Review by lucianoeli — APPROVED** [review]

File: app/src/test/java/com/example/cleanarchitecture/viewmodel/CharacterViewModelTest.kt [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        Dispatchers.setMain(testDispatcher)
+        useCase = CharacterUseCaseImpl(characterRepository)
+        viewModel = CharacterViewModel(useCase)
+    }
+
+    @After
+    fun `after`() {
+        Dispatchers.resetMain()
+        testDispatcher.cleanupTestCoroutines()
+    }
```
Comment (nicolbidone) [inline app/src/test/java/com/example/cleanarchitecture/viewmodel/CharacterViewModelTest.kt:65]: if you repeat this in each viewmodel test you can move the functionality to a parent class BaseViewModelTest and extend from it

File: app/src/test/java/com/example/cleanarchitecture/viewmodel/CharacterInformationViewModelTest.kt [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    fun `set up`() {
+        Dispatchers.setMain(testDispatcher)
+        viewModel = CharacterInformationViewModel()
+    }
+
+    @After
+    fun `after`() {
+        Dispatchers.resetMain()
+        testDispatcher.cleanupTestCoroutines()
+    }
```
Comment (nicolbidone) [inline app/src/test/java/com/example/cleanarchitecture/viewmodel/CharacterInformationViewModelTest.kt:46]: if you repeat this in each viewmodel test you can move the functionality to a parent class BaseViewModelTest and extend from it

File: app/src/test/java/com/example/cleanarchitecture/viewmodel/CharacterViewModelTest.kt (hunk 1 above)
Comment (ezequielsetzesglobant) [reply app/src/test/java/com/example/cleanarchitecture/viewmodel/CharacterViewModelTest.kt:65]: done

File: app/src/test/java/com/example/cleanarchitecture/viewmodel/CharacterInformationViewModelTest.kt (hunk 2 above)
Comment (ezequielsetzesglobant) [reply app/src/test/java/com/example/cleanarchitecture/viewmodel/CharacterInformationViewModelTest.kt:46]: done

**Review by nicolbidone — APPROVED** [review]

**Pull request merged**